packages/solana/test/models_test.dart

```diff
@@ -0,0 +1,165 @@
+// ignore_for_file: prefer-first
+
+import 'package:solana/dto.dart';
+import 'package:solana/solana.dart';
+import 'package:test/test.dart';
+
+import 'config.dart';
+
+void main() {
+  late final SolanaClient solanaClient;
+  late final Wallet source;
+  late final Wallet destination;
+  late Mint token;
+
+  setUpAll(() async {
+    source = await Ed25519HDKeyPair.random();
+    solanaClient = createTestSolanaClient();
+    destination = await Ed25519HDKeyPair.random();
+    // Add tokens to the sender
+    await solanaClient.requestAirdrop(
+      lamports: 100 * lamportsPerSol,
+      address: source.publicKey,
+      commitment: Commitment.confirmed,
+    );
+    token = await solanaClient.initializeMint(
+      mintAuthority: source,
+      decimals: 2,
+      commitment: Commitment.confirmed,
+    );
+    final associatedAccount = await solanaClient.createAssociatedTokenAccount(
+      mint: token.address,
+      funder: source,
+      commitment: Commitment.confirmed,
+    );
+
+    await solanaClient.mintTo(
+      destination: Ed25519HDPublicKey.fromBase58(associatedAccount.pubkey),
+      amount: _tokenMintAmount,
+      mint: token.address,
+      authority: source,
+      commitment: Commitment.confirmed,
+    );
+  });
+
+  group('SplTokenTransferCheckedInfo test', () {
+    test('authority is not available', () {
+      // TODO
+    });
+
+    test('authority is available', () async {
+      final wallet = await Ed25519HDKeyPair.random();
+      final programAccount = await solanaClient.createAssociatedTokenAccount(
+        mint: token.address,
+        funder: source,
+        owner: wallet.publicKey,
+        commitment: Commitment.confirmed,
+      );
+      final senderTokenAccounts =
+          await solanaClient.rpcClient.getTokenAccountsByOwner(
+        source.address,
+        const TokenAccountsFilterByProgramId(TokenProgram.programId),
+        encoding: Encoding.jsonParsed,
+        commitment: Commitment.confirmed,
+      );
+
+      await solanaClient.sendAndConfirmTransaction(
+        message: Message.only(
+          TokenInstruction.transferChecked(
+            amount: 10,
+            decimals: 2,
+            source:
+                Ed25519HDPublicKey.fromBase58(senderTokenAccounts[0].pubkey),
+            mint: token.address,
+            destination: wallet.publicKey,
+            owner: source.publicKey,
+          ),
+        ),
+        signers: [source],
+        commitment: Commitment.finalized,
+      );
```

Comment:
It keeps failing with 
```
jsonrpc-2.0 error (-32002): Transaction simulation failed: Attempt to debit an account but found no record of a prior credit.
	{accounts: null, err: AccountNotFound, logs: [], unitsConsumed: 0}
```

I assume I don't pass correct params or I'm missing something.